### DISCUSSION TITLE
Drawer content should be scrollable

### DISCUSSION THREAD

**Discussion by italodeandra** [mantinedev/mantine#2483] [Feature requests]
### What package has an issue

@mantine/core

### Describe the bug

Can't scroll down when the drawer content doesn't fit the window height.

![image](https://user-images.githubusercontent.com/19225266/190839477-d777e8a0-84af-4a6c-b329-cbce5d7779fe.png)

And it has been very difficult to make it work with ScrollArea.

My suggestion would be to make a new <Drawer.Content> that is scrollable and also create a <Drawer.Actions> so you can have a button stick to the bottom of the Drawer and let only the content be scrollable (so the Save button is always visible for example).

### What version of @mantine/hooks page do you have in package.json?

5.3.2

### If possible, please include a link to a codesandbox with the reproduced problem

https://mantine.dev/core/drawer/

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
It is in the backlog, this feature is panned to be added

**Reply by italodeandra** [reply]
Nice, thanks!

**Reply by mvcodelabs** [reply]
After half an hours, i solved it like this. The main thing that kept this from working was `min-height: 0` on the body of the drawer. [source (stackoverflow)](https://stackoverflow.com/questions/36230944/prevent-flex-items-from-overflowing-a-container/66689926#66689926)

```jsx
<Drawer
  styles={(theme) => ({
    drawer: {
      display: "flex",
      flexDirection: "column",
      height: "100%",

      "& > :nth-child(2)": {
        flex: 1,
        display: "flex",
        flexDirection: "column",
        minHeight: 0, // VERY IMPORTANT, idk why but it works
      },
    },
  })}
>
  <Box sx={{ flexGrow: 1, overflow: 'auto' }}>
    {/* scrollable content */}
  </Box>
  <Box sx={{ flexShrink: 0 }}>
    {/* bottom actions */}
  </Box>
</Drawer>
```

**Reply by ksdme** [reply]
This simpler configuration worked for me. Although, it assumes that the Drawer has exactly one child.

```tsx
<Drawer
  styles={{
    drawer: {
      display: "flex",
      flexDirection: "column",
    },
    body: {
      flexGrow: 1,
      overflowY: "auto",
    },
  }}
>
</Drawer>
```

**Comment by ibrahimbutt** [comment]
@rtivital is this in the works? If so, do you have an ETA?

**Reply by rtivital** [reply]
It is in the backlog, no ETA
